### PR TITLE
VACMS-2872: Learning Center category AX

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #2872 

## Testing done
Visual / behat

## Screenshots
image

## QA steps
@oksana-c to qa manually (please).

### CODE REVIEW

File: tests/behat/drupal-spec-tool/views.feature [hunk 1]
```diff
@@ -187,3 +186,8 @@ Feature: Views
 | Services | Facility health services | block_1 | Block |
 | Services | VAMC system health services | block_2 | Block |
 | Services | Master | default | Master |
+| Recent content | Master | default | Master |
```
Comment (swirtSJW) [inline tests/behat/drupal-spec-tool/views.feature:189]: sort seems off

**Review by swirtSJW — COMMENTED** [review]
Just a couple Q related to behat.

File: tests/behat/features/save_and_continue.feature [hunk 2]
```diff
@@ -4,49 +4,49 @@ Feature: Save and continue button works as expected.
...
-    Then I should see "test button label - edited"
-    Then I should see "Behat save and continue new test section header 1"
-    Then I should see "Behat save and continue new test section header 2 - edited"
-    Then I should see "Behat save and continue new test checklist item 1"
-    Then I visit the "edit" page for a node with the title "Behat save and continue new test - edited"
-
-    # Make sure we are in draft state
-    Then the "draft" option from "#edit-moderation-state-0-state" should be selected
+#@save_and_continue
+#  Scenario: Log in, edit, and save nodes with save and continue button and confirm revision saves changes.
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply tests/behat/features/save_and_continue.feature:8]: Note above, @swirtSJW

File: tests/behat/drupal-spec-tool/views.feature (hunk 1 above)
Comment (ethanteague) [reply tests/behat/drupal-spec-tool/views.feature:189]: Just a quick note to confirm that the weird sort isn't coming from a weird rebase - just pulled in from the sheet prior to beginning any mod's, and the order matches what we are seeing above, @swirtSJW

**Comment by ethanteague** [comment]
Updated, and ready for code review, @swirtSJW @oksana-c

**Review by swirtSJW — APPROVED** [review]

**Comment by swirtSJW** [comment]
Code looks fine, but this UI seems confusing
image

Are they dealing with categories or topic?   There ought to be alignment between these two.

**Review by oksana-c — COMMENTED** [review]
@ethanteague , please check updated spec in #2872 . Also, agreed w/ @swirtSJW "Add topics" should read "Add categories"

**Comment by ethanteague** [comment]
> @ethanteague , please check updated spec in #2872 . Also, agreed w/ @swirtSJW "Add topics" should read "Add categories"

Thanks, @oksana-c & @swirtSJW  :+1: - will update

**Comment by ethanteague** [comment]
@oksana-c @swirtSJW hoping we can break out the no repeats allowed in Entity browser requirement into a separate task. I tried a bunch of patches, and none fit the bill. I tried some manual tinkering to determine LOE, and didn't get very far. All of this is to say that the duplicate prevention thing is gonna take a while, and hoping we can get this task over the finish line without having to address.

File: config/sync/entity_browser.browser.topics.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+uuid: 3483f30e-c910-4c67-a5e5-405be5dd31cc
+langcode: en
+status: true
+dependencies:
+  config:
+    - views.view.taxonomy_entity_browsers
+  module:
+    - views
+name: topics
```
Comment (oksana-c) [inline config/sync/entity_browser.browser.topics.yml:9]: `name: topics`  > `name: categories`

also, the name of this config file contains "topics" and should be updated to match new term

File: config/sync/core.entity_form_display.node.checklist.default.yml [hunk 4]
```diff
@@ -4,6 +4,7 @@ status: true
 dependencies:
   config:
     - entity_browser.browser.lc_benefit_hubs
+    - entity_browser.browser.topics
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.node.checklist.default.yml:7]: `browser.topics`  will change for each field dependency once the browser is renamed

File: config/sync/core.entity_form_display.node.checklist.default.yml [hunk 5]
```diff
@@ -12,13 +13,16 @@ dependencies:
     - field.field.node.checklist.field_description
     - field.field.node.checklist.field_intro_text_limited_html
     - field.field.node.checklist.field_lc_benefit_hubs
+    - field.field.node.checklist.field_lc_other_categories
+    - field.field.node.checklist.field_lc_primary_category
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.node.checklist.default.yml:17]: I'll be a bearer of dreary news this morning... unfortunately.
Last Friday we learned that the Learning Center may in fact not be called Learning Center anymore. One potential name is "Support & Resources". That being said, it's fine for us to keep field Labels as is for now containing "Learning Center", but it's best to keep any machine names without specific `lc` prefix. 
Let's change all these fields to `field_primary_category` and `field_other_categories`. 
Sorry for all these change requests @ethanteague

File: config/sync/core.entity_form_display.node.checklist.default.yml [hunk 6]
```diff
@@ -95,6 +99,21 @@ third_party_settings:
         required_fields: false
       label: 'Include Alert'
       region: content
+    group_learning_center_categories:
```
Comment (oksana-c) [inline config/sync/core.entity_form_display.node.checklist.default.yml:102]: per comment above `group_learning_center_categories` > `group_categories`

**Review by oksana-c — CHANGES_REQUESTED** [review]

**Comment by oksana-c** [comment]
in regards to preventing duplicates in "Other categories" field - I'll open a follow up story. @kaise-lafrai is adding a contrib for duplicate validation in #2941 . Once that one is merged, we'll be able to add duplicate constraint to "other categories" field separately. Thanks @ethanteague

File: config/sync/core.entity_form_display.node.checklist.default.yml (hunk 5 above)
Comment (ethanteague) [reply config/sync/core.entity_form_display.node.checklist.default.yml:17]: Np a'tall, @oksana-c - easy to change.

**Comment by oksana-c** [comment]
@ethanteague , thanks for refactoring the fields. I reviewed and made some tweaks to field widgets. Also, added new fields to a new content type that was merged today "Learning Center Detail Page".
Updated all tests as well.

This branch is not building in Devshop .. when you see this, please pull the updates locally and open a new PR w/ all this work so we can verify tests pass in the morning and deploy tomorrow afternoon. Thanks so much!

**Comment by ethanteague** [comment]
> @ethanteague , thanks for refactoring the fields. I reviewed and made some tweaks to field widgets. Also, added new fields to a new content type that was merged today "Learning Center Detail Page".
> Updated all tests as well.
> 
> This branch is not building in Devshop .. when you see this, please pull the updates locally and open a new PR w/ all this work so we can verify tests pass in the morning and deploy tomorrow afternoon. Thanks so much!

Pulled this down, and trying to rebase off of master with your commits is creating a bunch of conflicts that are going to be difficult to resolve, @oksana-c . Wondering if this is the reason for the build failure?

**Comment by oksana-c** [comment]
closing in favor of #2963

**Pull request closed without merging**